### PR TITLE
infoschema: set the capacity of sieve to 0 for disabling infoschema v2

### LINKED ISSUE #55490: Infoschema V2 Cache Size metric is unchanged when set tidb_schema_cache_size 0
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

set @@tidb_schema_cache_size =0
img_v3_02dt_e9df03fa-ef37-4831-8756-056d08303b1g

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #55490

Problem Summary:

In the previous implementation, if the setting value is 0, we would not call `SetCapacity` for sieve, nor `onUpdateLimit`.

### What changed and how does it work?

always call `onUpdateLimit` for setting any capacity value

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Now `set @@tidb_schema_cache_size = 0` can update the metric

image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/infoschema/sieve.go [hunk 1]
```diff
@@ -46,10 +46,11 @@ func (t *entry[K, V]) Size() uint64 {
...
 type Sieve[K comparable, V any] struct {
-	ctx      context.Context
-	cancel   context.CancelFunc
-	mu       sync.Mutex
-	size     uint64
+	ctx    context.Context
+	cancel context.CancelFunc
+	mu     sync.Mutex
+	size   uint64
+	// capacity can be set to zero for indicating unlimited capacity.
```
Comment (tangenta) [inline pkg/infoschema/sieve.go:53]: Zero means infoschema v2 is disabled

**Review by tangenta — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**